vpr/src/place/simpleRL_move_generator.cpp

```diff
@@ -58,6 +28,60 @@ void SimpleRLMoveGenerator::process_outcome(double reward, e_reward_function rew
  *  K-Armed bandit agent implementation   *
  *                                        *
  *                                        */
+KArmedBanditAgent::KArmedBanditAgent(size_t num_moves, e_agent_space agent_space)
+    : num_available_moves_(num_moves)
+    , propose_blk_type_(agent_space == e_agent_space::MOVE_BLOCK_TYPE) {
+    std::vector<int> available_logical_block_types = get_available_logical_blk_types_();
+    num_available_types_ = available_logical_block_types.size();
+
+    num_available_actions_ = propose_blk_type_ ? (num_available_moves_ * num_available_types_) : num_available_moves_;
+
+    action_logical_blk_type_.clear();
+
+    for (auto logical_blk_type_idx : available_logical_block_types) {
+        action_logical_blk_type_.push_back(logical_blk_type_idx);
+    }
+}
+
+e_move_type KArmedBanditAgent::action_to_move_type_(const size_t action_idx) {
+    e_move_type move_type = e_move_type::INVALID_MOVE;
+
+    if (action_idx < num_available_actions_) {
+        move_type = (e_move_type)(action_idx % num_available_moves_);
+    }
+
+    return move_type;
+}
+
+int KArmedBanditAgent::action_to_blk_type_(const size_t action_idx) {
+    if (propose_blk_type_ && action_idx < num_available_actions_) {
+        return action_logical_blk_type_[action_idx / num_available_moves_];
+    } else {
+        return -1;
```

Comment:
Is -1 also INVALID_MOVE?  If so, better to return INVALID_MOVE.  If not, why have two invalid codes?

Reply:
If logical block type index is set to -1, the block type would be selected randomly. I'll add a short comment to clarify this.